lib/ansible/modules/database/postgresql/pg_hba.py

```diff
@@ -0,0 +1,598 @@
+    for k in keys:
+        if not rule[k]:
+            del rule[k]
+
+
+def rule2source_ipnw(rule):
+    '''
+    This method is used to get the source of a rul as an ipaddress object if possible.
+    '''
+    if '/' not in rule['src'] and 'mask' in rule.keys():
+        if ':' in rule['mask']:
+            # ipaddress module cannot work with ipv6 netmask, so lets convert it to prefixlen
+            binvalue = "{0:b}".format(int(ipaddress.IPv6Address(rule['mask'])))
+            if '01' in binvalue:
+                msg = 'IPv6 mask {} contains 1 after 0 and cannot be converted to prefixlen'
+                raise ValueError(msg.format(rule['mask']))
+            prefixlen = binvalue.count('1')
+            sourcenw = '{0}/{1}'.format(rule['src'], prefixlen)
+        else:
+            sourcenw = '{0}/{1}'.format(rule['src'], rule['mask'])
+    else:
+        sourcenw = rule['src']
+
+    try:
+        return ipaddress.ip_network(sourcenw, strict=False)
+    except ValueError:
+        return rule['src']
+
+
+def rule2key(rule):
+    '''
+    This method can be used to get the ey from a rule.
+    '''
+    if rule['type'] == 'local':
+        source = 'local'
+    else:
+        source = str(rule2source_ipnw(rule))
+    return (source, rule['db'], rule['usr'])
+
+
+def new_rules(contype, databases, users, source, netmask, method, options):
+    '''
+    This function can be called with a comma seperated list of databases and a comma seperated list
+    of users and it will act as a generator that returns a expanded list of rules one by one.
+    '''
+    if method not in PG_HBA_METHODS:
+        msg = "invalid method {0} (should be one of '{1}')."
+        raise PgHbaError(msg.format(method, "', '".join(PG_HBA_METHODS)))
+    if contype not in PG_HBA_TYPES:
+        msg = "invalid connection type {0} (should be one of '{1}')."
+        raise PgHbaError(msg.format(contype, "', '".join(PG_HBA_TYPES)))
+    # Add the job
+    rule = dict(zip(PG_HBA_HDR, [contype, databases, users, source, netmask, method, options]))
+
+    for database in databases.split(','):
+        for usr in users.split(','):
+
+            rule = dict(zip(PG_HBA_HDR, [contype, database, usr, source, netmask, method, options]))
+
+            if contype == 'local':
+                del rule['src']
+                del rule['mask']
+            elif '/' in source:
+                del rule['mask']
+            else:
+                rule['src'] = str(rule2source_ipnw(rule))
+                del rule['mask']
+
+            clean_empty_rule_keys(rule)
+
+            line = [rule[k] for k in PG_HBA_HDR if k in rule.keys()]
+            rule['line'] = "\t".join(line)
+            yield rule
+
+
+def main():
+    '''
+    This function is the main function of this module
+    '''
+    module = AnsibleModule(
+        argument_spec=dict(
+            contype=dict(default=None, choices=PG_HBA_TYPES + ['']),
+            backup=dict(default=False, type='bool'),
```

Comment:
`backup` is already a file common arguments ([see](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L240)).